content/blog/streamline-and-optimize-ml-workflows-with-hpe-ezmeral-unified-analytics.md

```diff
@@ -0,0 +1,153 @@
+
+UI and APIs: MLflow makes it easy to use and integrate programmatically by offering a web-based user interface as well as APIs. 
+
+**Experiments**
+
+An experiment is registered in the MLflow tracking service, and different runs are carried out with different parameters and features. The parameters can then be logged into MLflow and viewed on the UI. Different runs are then compared so that the best metric can be chosen and moved into production. 
+
+To begin, allow the experiment to access MLflow from the Jupyter notebook. Then as we have used TensorFlow LSTM as our ML model, we use mlflow.tensorflow.autolog() to log automatically the training parameters, hyperparameters and model metrics.
+
+![](/img/figure-1a-experiment.png "Figure 2. Creating an experiment with auto logging. ")
+
+Once the model is ready, the run details are available via MLflow APIs like search_run(). The best run can be auto-selected as per our requirements.
+
+![](/img/fig-1b-best-run-experiment.png "Figure 3. Selecting the best run of the experiment and loading the model.")
+
+On the Web UI for MLflow, we can access multiple runs of the ML experiment for different parameter combinations. Details of the run can be accessed by clicking on the corresponding run name.
+
+![](/img/fig-2-multiple-runs.png "Figure 4. Results from multiple runs made on the experiment with different parameters in each run. ")
+
+Model performance can be obtained by comparing various runs. In the image below, all six runs have been selected and we see that the model with epochs=20, performs best with the least loss and mean absolute error (MAE).
+
+![](/img/fig-3-multiple-runs-compared.png "Figure 5. Comparison of all runs based on a relevant metric. ")
+
+Now that the best model has been identified, it can be registered programmatically or through the MLflow Web UI. Once registered, the model can be deployed to the appropriate staging-production region. 
+
+![](/img/fig-4-model-registry-different-models.png "Figure 6. Model registry with different models. ")
+
+**Code Repository**
+
+A code repository is a storage location for code and other software development assets, such as documentation, tests, and scripts. They are often used to manage and organize a software project's codebase and collaborate with other project developers.
+
+The source code for the model is stored in a private repository on GitHub then seamlessly pulled into the IDE using git commands. The updated model can be pushed to different branches of the source repo. 
+
+![](/img/code-repository.png "Figure 7.  Source code for models is stored on Github in a private repository. ")
+
+**Model Deployment**
+
+The next step is to deploy the best model chosen from the MLflow experiment. Here we are using KServe which is a standard Model Inference Platform on Kubernetes, built for highly scalable use cases.
```

Comment:
Change the second sentence to read:
Here, Kserve, which is a standard model inference platform on Kubernetes that is build for highly scalable use cases, is being used. (I know... many prefer the active tense, but sometimes it's hard to get around the Royal We.